File: src/main/kotlin/com/elissamc/actions/TeleportAction.java
package com.elissamc.actions;


import cn.nukkit.Player;
import cn.nukkit.level.Location;


/**
 * Inventory action that teleports player to specified location.
 *
 */
public class TeleportAction implements Action {
    /**
     * Target location.
     */
    private final Location location;

    /**
     * Creates a new action.
     *
     * @param location
     */
    public TeleportAction(final Location location) {
        this.location = location;
    }

    @Override
    public void execute(final Player player) {
        // Just teleport player to target location.
        player.teleport(this.location);
    }
}

File: src/main/kotlin/com/elissamc/actions/TeleportAction.kt
package com.elissamc.actions


import cn.nukkit.Player
import cn.nukkit.level.Location


/**
 * Inventory action that teleports player to specified location.
 *
 */
class TeleportAction
/**
 * Creates a new action.
 *
 * @param location
 */
(
        /**
         * Target location.
         */
        private val location: Location) : Action {

    override fun execute(player: Player) {
        // Just teleport player to target location.
        player.teleport(this.location)
    }
}